idaes/core/scaling/custom_scaler_base.py

```diff
@@ -567,55 +566,47 @@ def call_submodel_scaler_method(
         default scaler for the submodel is used.
 
         Args:
-            model: parent model of submodel
-            submodel: local name of submodel to be scaled as str
+            submodel: submodel to be scaled
             submodel_scalers: user provided ComponentMap of Scalers to use for submodels
             method: name of method to call from submodel (as string)
             overwrite: whether to overwrite existing scaling factors
 
         Returns:
             None
         """
-        # Get actual submodel object from name
-        # For this method, we have to use the component name as the Scaler is written
-        # before the model is constructed.
-        sm_obj = model.find_component(submodel)
-
         if submodel_scalers is None:
             submodel_scalers = {}
 
         # Iterate over indices of submodel
-        for smdata in sm_obj.values():
+        for smdata in submodel.values():
             # Get Scaler for submodel
-            if sm_obj in submodel_scalers:
-                scaler = submodel_scalers[sm_obj]
+            if submodel in submodel_scalers:
+                scaler = submodel_scalers[submodel]
                 if callable(scaler):
                     # Check to see if Scaler is callable - this implies it is a class and not an instance
                     # Call the class to create an instance
                     scaler = scaler()
-                _log.debug(f"Using user-defined Scaler for {model}.{submodel}.")
+                _log.debug(f"Using user-defined Scaler for {submodel}.")
```

Comment:
Is this going to give human readable output, or is it going to be some Pyomo thing in brackets?

Consider using `submodel.name`

Reply:
It will end up being `submodel.name` - it is not obvious but the f-string magic casts it to a string.